aten/src/ATen/cuda/CUDABlas.h

```diff
@@ -19,6 +19,25 @@ namespace at {
 namespace cuda {
 namespace blas {
 
+// RAII guard that sets the CuBLAS pointer mode and restores it to
+// its previous value when the guard is destroyed
+class PointerModeGuard {
+public:
+  PointerModeGuard(cublasHandle_t handle, cublasPointerMode_t mode) :
+      handle(handle) {
+    TORCH_CUDABLAS_CHECK(cublasGetPointerMode(handle, &previous_mode));
+    TORCH_CUDABLAS_CHECK(cublasSetPointerMode(handle, mode));
+  }
+
+  ~PointerModeGuard() {
+    cublasSetPointerMode(handle, previous_mode);
```

Comment:
I'm not sure what to do here if `cublasSetPointerMode()` fails. Before, I had it wrapped with `TORCH_CUDABLAS_CHECK()`, but the CI job `pytorch_linux_xenial_cuda10_2_cudnn7_py3_gcc7_build` complained `error: throw will always call terminate() [-Werror=terminate]`.

I never realized before that there was an issue with throwing exceptions in destructors, but it makes sense. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Re-never-fail

So I'm not sure if we should just ignore the error code from `cublasSetPointerMode`, or if we should do something else.

Reply:
Yup, you're supposed to ignore it.